src/init.sh

```diff
@@ -170,5 +180,173 @@ function init_help()
     '  init --arch <arch> - Set the arch field in the kworkflow.config file.' \
     '  init --remote <user>@<ip>:<port> - Set remote fields in the kworkflow.config file.' \
-    '  init --target <target> Set the default_deploy_target field in the kworkflow.config file'
+    '  init --target <target> - Set the default_deploy_target field in the kworkflow.config file' \
+    '  init --interactive - Perform first time setup in an interactive manner. Recommended for new users.'
+}
+
+# This function gets four configuration values from git: user name, user email,
+# editor, and initial branch name and stores them inside an associative array
+# for future access. Checks if these values are empty and store this information
+# in a boolean variable inside the array.
+#
+# @git_config: associative array received as a nameref attribute to store all
+#              values acquired in this function.
+#
+# Returns: git_config[configured]: 1, if there is any revelant information in
+#          git missing, and 0, if git is fully configured.
+function get_git_config()
+{
+  local -n git_config="$1"
+
+  git_config['name']=$(git config user.name)
+  git_config['email']=$(git config user.email)
+  git_config['editor']=$(git config core.editor)
+  git_config['branch']=$(git config init.defaultBranch)
+
+  git_config['configured']=0
+  if [[ -z "${git_config['name']}" ||
+    -z "${git_config['email']}" ||
+    -z "${git_config['editor']}" ||
+    -z "${git_config['branch']}" ]]; then
+    git_config['configured']=1
+  fi
+}
+
+# This function sets if the scope of the git configuration will be
+# local or global.
+#
+# @config_cmd: command in which will be added the configuration scope.
+#
+# Returns: the git config command, adjusted if the scope is local or global
+function set_git_config_scope()
+{
+  local command
+  local scope
+
+  command='git config'
+
+  if is_inside_work_tree; then
+    printf '\nSelect the scope of this configuration:\n' 1>&2
+
+    select scope in 'Local' 'Global'; do
+      case $scope in
+        'Global')
+          command+=' --global'
+          break
+          ;;
+        'Local')
+          command+=' --local'
+          break
+          ;;
+      esac
+    done
+  else
+    warning 'You are not in a git repository. All modifications will be global' 1>&2
+    if [[ $(ask_yN 'Would you like to proceed?' 'y') =~ '1' ]]; then
+      command+=' --global'
+    else
+      command=''
+    fi
+  fi
+
+  printf '%s\n' "$command"
+}
+
+# Allows the user to set the empty configurations if any.
+#
+# @config_git: associative array with data about git configuration.
+#
+# Returns: config_git: array with the selected information configured on git
+function set_git_config()
+{
+  local -n config_git="$1"
+  local flag="$2"
+  local git_editor_suggestion
+  local config_cmd
+  local scope
+
+  flag=${flag:-'SILENT'}
+
+  if [[ "${config_git['configured']}" =~ '1' ]]; then
+    printf '\nNow Git will be configured.\n' 1>&2
```

Comment:
I did not get why we need `1>&2` here.

Reply:
Also, in the following lines, you have a check and return, which makes this message outside context.

Reply:
> I did not get why we need `1>&2` here.

We redirect the standard output to the standard error because these messages are for user instruction only. Redirecting to the standard error (which is already done to deal with the select messages), we can easily disregard these messages in the test file.

Reply:
> Also, in the following lines, you have a check and return, which makes this message outside context.

Return only will be executed if the user chooses to cancel the operation in case it is not in a git repository and doesn't want to make global changes in git, so, even if the return statement be executed, there will be some context.